### PR TITLE
[vulkan] Add target triple for Apple M1 GPUs

### PR DESCRIPTION
**Pull request by antiagainst**
This commit add architecture/product/OS registrations for Apple
M1 GPUs to allow targeting them better.

### CODE REVIEW

**Comment by antiagainst** [comment]
Depends on https://reviews.llvm.org/D122414

**Review by MaheshRavishankar — APPROVED** [review]

File: iree/compiler/Dialect/Vulkan/Utils/TargetTriple.cpp [hunk 1]
```diff
@@ -207,6 +222,30 @@ CapabilitiesAttr getCapabilities(const TargetTriple &triple,
       storageBuffer8BitAccess = true, storagePushConstant8 = true;
       uniformAndStorageBuffer8BitAccess = true;
 
+      variablePointers = variablePointersStorageBuffer = true;
+      break;
+    case TargetTripleArch::Apple_M1:
+      // Example: https://vulkan.gpuinfo.org/displayreport.php?id=10312
```
Comment (powderluv) [inline iree/compiler/Dialect/Vulkan/Utils/TargetTriple.cpp:228]: nit: this refers to a MALI devices. We probably want to update to 
https://vulkan.gpuinfo.org/displayreport.php?id=14522

**Review by powderluv — APPROVED** [review]
minor nit. otherwise LGTM

**Review by ThomasRaoux — APPROVED** [review]

File: iree/compiler/Dialect/Vulkan/Utils/TargetTriple.cpp (hunk 1 above)
Comment (antiagainst) [reply iree/compiler/Dialect/Vulkan/Utils/TargetTriple.cpp]: Oops. Good catch! I thought I changed the URL. But somehow messed up. Fixed.

**Pull request merged**